operator/pkg/sidecarinjector/sidecarinjector.go

```diff
@@ -364,8 +366,8 @@ func (m *SidecarInjector) adaptAppContainerSpec() error {
 func (m *SidecarInjector) injectSidecarContainer() {
 	sidecarContainer := corev1.Container{
 		Name:            sidecarContainerName,
-		Image:           m.completeImageURL(sidecarContainerImageName(m.SidecarImageName, m.Runtime)),
-		ImagePullPolicy: corev1.PullPolicy(m.ImagePullPolicy),
+		Image:           m.completeImageURL(sidecarContainerImageName(m.meshService.SidecarImage, m.dynamicSpec.spec())),
+		ImagePullPolicy: corev1.PullPolicy(m.dynamicSpec.spec().ImagePullPolicy),
 		Command:         sidecarContainerCmd,
 		VolumeMounts:    sidecarContainerVolumeMounts,
 		Env:             siecarContainerEnvs,
```

Comment:
Nit: siecarContainerEnvs is a typo (didn't related to the change)

Reply:
Fixed.